0.3/freefare/key-deriver.go

```diff
@@ -0,0 +1,45 @@
+package freefare
+
+// #include <freefare.h>
+import "C"
+
+type MifareKeyType int
+
+const (
+	MIFARE_KEY_DES MifareKeyType = iota
+	MIFARE_KEY_2K3DES
+	MIFARE_KEY_3K3DES
+	MIFARE_KEY_AES128
+)
+
+type MifareKeyDeriver struct {
```

Comment:
Some documentation would be good here.  If you don't want to add any, I can try to cobble something together.  Have not been working on this for a long time though.

Reply:
Documentation for key deriver is kind of hard to do, since it's just a wrapper, and the original doesn't have documentation either. The whole process for adding it was described [here](https://github.com/nfc-tools/libfreefare/issues/77), the algorithm itself is decribed [here](https://www.nxp.com/docs/en/application-note/AN10922.pdf) and the commit that initially added it is [here](https://github.com/nfc-tools/libfreefare/commit/9d88c18833450a91b97fc6ac91c1d983ed3a8681). New helper functions in ultralight.go are just wrappers on [this](https://github.com/nfc-tools/libfreefare/blob/master/examples/mifare-ultralightc-diversify.c) example from libnfc repo. To be honest, I'm not that fluent in C to be able to succinctly describe what exactly happens in `Begin`, `UpdateUID` and `End` functions just by looking at code.